### PR TITLE
add Plot.shuffle

### PR DESCRIPTION
**Pull request by mbostock**
I was confused why {sort: Math.random} didn’t work, but it’s because Math.random.length is zero, and so it’s interpreted as a comparator rather than an accessor. 😞 But anyway, now you can say Plot.shuffle(_options_) which seems nice.

### CODE REVIEW

File: src/transforms/sort.js [hunk 1]
```diff
@@ -2,6 +2,10 @@ import {ascendingDefined} from "../defined.js";
 import {valueof} from "../mark.js";
 import {basic} from "./basic.js";
 
+export function shuffle(options) {
+  return basic(options, sortValue(Math.random));
+}
```
Comment (Fil) [inline src/transforms/sort.js:7]:
```suggestion
import {basic} from "./basic.js";
import {randomUniform, randomLcg} from "d3";

export function shuffle({seed, ...options}) {
  return basic(options, sortValue(randomUniform.source(randomLcg(seed))()));
}
```

**Review by Fil — CHANGES_REQUESTED** [review]
Do we have a test plot where it's useful?

For the documentation, just after Plot.sort:

~~~
### Plot.shuffle(*options*)

```js
Plot.shuffle({seed: 42, ...options})
```

Randomizes the data order. Pass a seed to make the shuffling reproducible.


~~~

File: src/transforms/sort.js (hunk 1 above)
Comment (Fil) [reply src/transforms/sort.js:7]: maybe
`sortValue((seed ? randomUniform.source(randomLcg(seed)) : randomUniform)())`

Comment (mbostock) [reply src/transforms/sort.js]: Okay, added a seed option. We don’t need randomUniform here; we can use Math.random directly.

**Review by Fil — APPROVED** [review]

**Pull request merged**